src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -47,6 +49,40 @@ public Optional<String> getDerivedTableName(UUID entityTypeId) {
       .fetchOptional(derivedTableNameField);
   }
 
+  public List<EntityTypeColumn> fetchNamesForSingleCheckbox(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getDerivedTableName(entityTypeId).get();
+    List<EntityTypeColumn> entityColumnsList = new ArrayList<>();
+
+    String requiredFieldName = "jsonb" + "->>'name'";
+    String abc =  "jsonb" + "->>'refId'";
+    String whereClauseName = "jsonb" + "->>'type'";
+    String customFieldType = "SINGLE_CHECKBOX";
+    ValueWithLabel valueWithLabelTrue = new ValueWithLabel().label("True").value("true");
+    ValueWithLabel valueWithLabelFalse = new ValueWithLabel().label("False").value("false");
+
+    Result<Record2<Object, Object>> result = jooqContext
+      .select(field(requiredFieldName), field(abc))
+      .from(sourceViewName)
+      .where(field(whereClauseName).eq(customFieldType))
+      .fetch();
+    result.stream()
+      .map(record -> {
+        Object value = record.get(0);
+        EntityTypeColumn entityTypeColumn = new EntityTypeColumn();
+        entityTypeColumn.name((value instanceof List) ?
+          ((List<?>) value).stream().map(item -> item != null ? item.toString() : null).collect(Collectors.joining(", "))
+          : (value != null ? value.toString() : null));
+        entityTypeColumn.dataType(new BooleanType());
+        entityTypeColumn.values(List.of(valueWithLabelTrue, valueWithLabelFalse));
+        entityTypeColumn.visibleByDefault(false);
+        entityTypeColumn.valueGetter("user_custom_fields.jsonb -> 'customFields' ->> 'refId'");
+        return entityTypeColumn;
+      })
+      .forEach(entityColumnsList::add);
+    return entityColumnsList;
```

Comment:
Rather than create a list, fill it, then return it, I'd do it all in 2 fluid chain of calls

So return the result of
```suggestion
      .toList();
```
and just remove `entityColumnsList`

Reply:
Actually, since the result is just getting added to another list, I'd just return the `Stream` directly and let the call decide what to do with it. That way, we avoid allocating and populating a list just to throw it away almost immediately; instead, the caller can just consume the stream directly without needing to allocate anything extra